src/lib/data/data_loader.ts

```diff
@@ -45,4 +51,45 @@ export class DataModel {
   async addBudget(budget: Budget): Promise<void> {
     return this.database.addDocument(Collections.Budgets, budget);
   }
+
+  //These should be moved
+  async addUser(email: string, password: string, user: User): Promise<void> {
```

Comment:
The `DataModel` class shouldn't directly call any firebase code.  Just as for storing things in the database, it uses the `Database` class rather than directly making calls to firestore, for user authentication, it should either make calls to an `Authenticator` class, or just entirely delegate the responsibility of authentication the `Authenticator` class, or something else that helps reduce the coupling between firebase and the `DataModel` class.

Reply:
Will resolve later